croquemort/migrations.py

```diff
@@ -113,3 +115,34 @@ def add_hash_prefixes(self):
         self._migrate_urls_list()
         for freq in ['hourly', 'daily', 'monthly']:
             self._migrate_frequency(freq)
+
+    @rpc
+    def migrate_urls_redirect(self):
+        """
+        [migration from 1.0.0 to 2.0.0]
+
+        Migrate all the url hash fields to the new schema adopted with the
+        "redirect handling" feature.
+
+        Plus, cleanup url hashes with no url field attached.
+
+        NB: should be idempotent
+        """
+        database = self.storage.database
+        for url_hash, data in self.storage.get_all_urls():
+            if data.get('checked-url'):
+                continue
+            if data.get('url'):
+                database.hset(url_hash, 'checked-url', data['url'])
+                database.hdel(url_hash, 'url')
+                # should we really be doing that? it can be misleading
```

Comment:
> We have ~80k urls in production. Considering ~5 sec for a check, it would take more than 100 hours to recheck everything :-/

I'm not that pessimist, given the asynchronous task it might take about a day BUT the risk is to be blacklisted from servers…